src/test/operators/typed_operator_base_test.hpp

```diff
@@ -0,0 +1,40 @@
+#include <boost/hana/for_each.hpp>
+
+#include "all_type_variant.hpp"
+#include "base_test.hpp"
+#include "constant_mappings.hpp"
+#include "storage/encoding_type.hpp"
+
+namespace opossum {
+
+// This is the base class for all tests that should run on every possible combination of data type and encoding.
+// A sample table is generate in TableScanBetweenTest. Depending on how other tests will use this, it might make sense
+// to have a single instantiation here.
+class TypedOperatorBaseTest : public BaseTestWithParam<std::tuple<DataType, EncodingType, bool /* nullable */>> {
+ public:
+  static std::string format(testing::TestParamInfo<ParamType> info) {
+    const auto& [data_type, encoding, nullable] = info.param;
+
+    return data_type_to_string.left.at(data_type) + encoding_type_to_string.left.at(encoding) +
+           (nullable ? "" : "Not") + "Nullable";
+  }
+};
+
+static std::vector<TypedOperatorBaseTest::ParamType> create_param_pairs() {
```

Comment:
minor: it's a triple/tuple, not a pair. I'd prefer just "create_params" ;)

Reply:
I find both versions to convey little meaning and changed it to `initialize_test_params`. Does that work for you?

Reply:
Thinking about it, it does not initialize anything. It's `create_test_params` now